src/frequenz/sdk/timeseries/formula_engine/_formula_generators/_fallback_formula_metric_fetcher.py

```diff
@@ -0,0 +1,55 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""FallbackMetricFetcher implementation that uses formula generator."""
+
+from frequenz.channels import Receiver
+
+from ... import Sample
+from ..._quantities import QuantityT
+from .. import FormulaEngine
+from .._formula_steps import FallbackMetricFetcher
+from ._formula_generator import FormulaGenerator
+
+
+# This is done as a separate module to avoid circular imports.
+class FallbackFormulaMetricFetcher(FallbackMetricFetcher[QuantityT]):
+    """A metric fetcher that uses formula generator.
+
+    The formula engine is generated lazily, meaning it is created only when needed.
+    Once the formula engine is initialized, it subscribes to its components
+    and begins calculating and sending the formula results.
+
+    To start formula engine, call `start` method or `fetch_next` method.
+    """
+
+    def __init__(self, formula_generator: FormulaGenerator[QuantityT]):
+        """Create a `FallbackFormulaMetricFetcher` instance.
+
+        Args:
+            formula_generator: A formula generator that generates formula
+                engine with fallback components.
+        """
+        super().__init__(name=formula_generator.namespace)
+        self._formula_generator: FormulaGenerator[QuantityT] = formula_generator
+        self._formula_engine: FormulaEngine[QuantityT] | None = None
+        self._receiver: Receiver[Sample[QuantityT]] | None = None
+
+    @property
+    def is_running(self) -> bool:
+        """Check whether the formula engine is running."""
+        return self._receiver is not None
+
+    def start(self) -> None:
+        """Initialize formula engine and start fetching samples."""
+        self._formula_engine = self._formula_generator.generate()  # type: ignore[assignment]
+        assert isinstance(self._formula_engine, FormulaEngine)
+        self._receiver = self._formula_engine.new_receiver()
```

Comment:
Mmm, interesting, but with this implementation `stop()` and `cancel()` will not work. I guess `cancel()` could still be a NOP, but you should probably override `stop()`.

Reply:
`stop()` should work because it has :
```
        if not self._tasks:
            return
```

Reply:
OK, I guess I reviewed this too quickly and didn't realized the missing `stop()` in the comment was this stop. I also missed that even when it had a `run()` method it didn't actually have any task running. I think I saw a `run()` and `stop()` and immediately assumed it had some task running.

I think this class is actually a good match for the new abstract `Service` interface, and a good case for splitting the `ServiceBase` with a basic implementation using tasks, as it doesn't really help in this case but it is nice to still have the common `Service` interface with start/stop/is_running/etc. for consistency.

Reply:
Ok, I will do that later today.

Reply:
sdk doesn't use frequenz-core-python now. Should I add it and use `Service`?
Or should I do that in separate PR?

Reply:
frequenz-core-python has no release, yet.
Should I add dependency to the v1.x.x? 

Maybe I could do that in separate PR (this one is pretty big)?

Reply:
No, sorry, I didn't mean to do it now, but I actually got thinking about this, and I wonder why isn't this just a `Receiver` with a couple of extra methods and we are already using `stop()` for some receivers (like in [`Merger`](https://github.com/frequenz-floss/frequenz-channels-python/blob/d8bafe8ee30f661ebe2b420c5fd9b240de049dd7/src/frequenz/channels/_merge.py#L131-L136).

Part of the reasons of splitting this to `core` is so we can use `core` in `channels` too, and maybe make these `Receiver`s a `Service` too.

So my suggestion would be to make it a `Receiver` (so `fetch_next()` would just be `receive()`) and follow the interface of `Service` so in the future when this can inherit from `Service` it is not a breaking change.

**OR**, you can leave things are they are and we take care of it when we introduce the dependency to `core`, which will probably need adapting many other classes :)

Reply:
This can't be Receiver, because stopping Receiver won't stop formula.
Here i would like to start/stop formula when needed. This means: subscribe/unsubscribe from all formula components, too.

Reply:
I still don't get this.

So there is no `stop()` for now (or there is one inherited from `BackgroundService` that doesn't do anything).

I guess in the future you want to add a `stop()`. You are saying this `stop()` should not stop the `self._formula_engine`? Or the `self._receiver`? Or both? In any case, if you make `FallbackMetricFetcher` a `Receiver`, why can't you just make `stop()` do whatever it should do? If it is a semantic issue, maybe you could call it `pause()` instead of `stop()` to better convey that it is not entirely stopped but just not producing values anymore temporarily?

I think it was also mentioned before, I find it confusing that we have this extra inheritance `FallbackFormulaMetricFetcher` -> `FallbackMetricFetcher`. Can't we get rid of `FallbackMetricFetcher`? What is it for? We already have some unnecessary inheritance in the SDK from the past, I would prefer not to introduce more unless it is really necessary.

Reply:
Done, I made it Receiver.